depends/class-2.6.0-a_max.patch

```diff
@@ -68,3 +68,98 @@ diff -uNr class_public-2.6.0.old/source/input.c class_public-2.6.0/source/input.
    /** - h (dimensionless) and [\f$ H_0/c\f$] in \f$ Mpc^{-1} = h / 2997.9... = h * 10^5 / c \f$ */
    class_call(parser_read_double(pfc,"H0",&param1,&flag1,errmsg),
               errmsg,
+
+diff -uNr class_public-2.6.0.old/source/thermodynamics.c class_public-2.6.0/source/thermodynamics.c
+--- class_public-2.6.0.old/source/thermodynamics.c	2017-08-22 13:46:49.000000000 -0400
++++ class_public-2.6.0/source/thermodynamics.c	2017-08-22 15:18:47.000000000 -0400
+@@ -2279,6 +2279,7 @@
+   double chi_heat;
+   int last_index_back;
+   double relative_variation;
++  double zend=1./pba->a_max - 1;
+
+   Yp = pth->YHe;
+
+@@ -2367,7 +2368,7 @@
+   /* initial step */
+   dz = dz_max;
+
+-  while (z > 0.) {
++  while (z > zend) {
+
+     class_test(dz < ppr->smallest_allowed_variation,
+                pth->error_message,
+@@ -2375,7 +2376,7 @@
+
+     /* - try next step */
+     z_next=z-dz;
+-    if (z_next < 0.) z_next=0.;
++    if (z_next < zend) z_next=zend;
+
+     class_call(thermodynamics_reionization_function(z_next,pth,preio,&xe_next),
+                pth->error_message,
+@@ -2660,7 +2661,7 @@
+   param.nH0 = 11.223846333047*param.obh2*(1.-param.Y);  /* number density of hydrogen today in m-3 */
+   param.fHe = param.Y/(1-param.Y)/3.97153;              /* abundance of helium by number */
+   param.zstart = ppr->recfast_z_initial; /* Redshift range */
+-  param.zend = 0.;
++  param.zend = 1./pba->a_max - 1.;
```

Comment:
Hmm maybe. All of the parts I changed the end of the z_table array was hard-coded to zero...so a_today didn't factor in previously. But maybe it should have?